tests/wamr-test-suites/wasi-test-script/run_wasi_tests.sh

```diff
@@ -33,27 +64,14 @@ else
         target_option="--target=i386"
     fi
 
-    # Run WASI thread proposal tests
+    # Run WASI thread tests in AOT mode
     exit_code=0
-    wasm_tests=$(ls tests/proposals/wasi-threads/*.wasm)
-    for test_wasm in ${wasm_tests}; do
-        test_aot="${test_wasm%.wasm}.aot"
-        test_json="${test_wasm%.wasm}.json"
-
-        echo "Compiling $test_wasm to $test_aot"
-        ${WAMRC_CMD} --enable-multi-thread ${target_option} \
-            -o $test_aot $test_wasm
-
-        echo "Running $test_aot"
-        expected=$(jq .exit_code ${test_json})
-        ${IWASM_CMD} $test_aot
-        ret=${PIPESTATUS[0]}
-
-        echo "expected=$expected, actual=$ret"
-        if [[ $expected != "" ]] && [[ $expected != $ret ]];then
-            exit_code=1
-        fi
-    done
+    thread_internal_tests=$(ls ${WAMR_DIR}/core/iwasm/libraries/lib-wasi-threads/test/*.wasm)
+    thread_internal_tests=($thread_internal_tests) # array to pass as arg to a function
+    run_aot_tests "${thread_internal_tests[@]}"
+    thread_proposal_tests=$(ls tests/proposals/wasi-threads/*.wasm)
+    thread_proposal_tests=($thread_proposal_tests) # array to pass as arg to a function
+    run_aot_tests "${thread_proposal_tests[@]}"
```

Comment:
Will we also run cases under
tests/c/testsuite/
tests/assemblyscript/testsuite
${WAMR_DIR}/core/iwasm/libraries/lib-socket/test?
Refer to interpreter mode test above

Reply:
Checked these testsuites with AOT, in each testsuite some of the tests are failing now so I wouldn't enable it in this PR, but I updated the script so it can be easily added by adding other test variables to the loop.